mount/mount_windows.go

```diff
@@ -90,8 +121,37 @@ func (m *Mount) GetParentPaths() ([]string, error) {
 
 // Unmount the mount at the provided path
 func Unmount(mount string, flags int) error {
+	mount = filepath.Clean(mount)
+
+	// Helpfully, both reparse points and symlinks look like links to Go
+	// Less-helpfully, ReadLink cannot return \\?\Volume{GUID} for a volume mount,
+	// and ends up returning the directory we gave it for some reason.
+	if mountTarget, err := os.Readlink(mount); err != nil {
+		// Not a mount point.
+		// This isn't an error, per the EINVAL handling in the Linux version
+		return nil
+	} else if mount != filepath.Clean(mountTarget) {
+		// Directory symlink
+		if err := bindUnmount(mount); err != nil {
+			return fmt.Errorf("failed to bind-unmount from %s: %w", mount, err)
+		}
+		return nil
+	}
```

Comment:
Comment's been decoupled, see https://github.com/containerd/containerd/pull/4419#discussion_r485615803

Reply:
This doesn't allow us to _avoid_ the ADS, as what we have there is the input to one one-way API (`hcsshim.GetLayerMountPath`) and we'd be comparing the output with another one-way API (`winio.DecodeReparsePoint`)'s result. So it's a sanity check to ensure the volume point we're unmounting points at the mounted volume for which we've stored the source path.

Reply:
This is actually the reason you're seeing the ```hcsshim::ActivateLayer failed in Win32: The process cannot access the file because it is being used by another process. (0x20): unknown``` error in comment https://github.com/containerd/containerd/pull/4419#issuecomment-1210930397

A ```bindUnmount()``` is run on the mount point, which just deletes the junction, leaving the layer still active.  Using ```winio.DecodeReparsePoint()```, with something like (see https://github.com/containerd/continuity/pull/208/files) this:

```go
info, err := os.Lstat(mount)
if err != nil {
	return fmt.Errorf("getting mount point info: %w", err)
}

if info.Mode()&os.ModeSymlink == 0 {
	// Not a mount point.
	return nil
}

rp, err := getReparsePoint(mount, info)
if err != nil {
	return fmt.Errorf("getting reparse point: %w", err)
}

if !rp.IsMountPoint {
	if err := bindUnmount(mount); err != nil {
		return fmt.Errorf("failed to bind-unmount from %s: %w", mount, err)
	}
	return nil
}
```

properly unmounts and the ```TestVolumeCopyUp``` test passes.